### ISSUE TITLE
Assertion when precompiling CSV in debug mode

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#28213]
```
$ ./usr/bin/julia-debug

julia> using CSV
[ Info: Precompiling module CSV

julia-debug: /home/keno/julia/src/dump.c:1890: jl_deserialize_typemap_entry: Assertion `((te->func.linfo->max_world == 0 && te->func.linfo->min_world == 1) || (te->func.linfo->max_world >= te->max_world && te->func.linfo->min_world <= te->min_world)) && "corrupt typemap entry structure"' failed.

signal (6): Aborted
in expression starting at /home/keno/.julia/packages/DataFrames/LwtB/src/DataFrames.jl:10
__libc_signal_restore_set at /build/glibc-itYbWN/glibc-2.26/signal/../sysdeps/unix/sysv/linux/nptl-signals.h:80 [inlined]
raise at /build/glibc-itYbWN/glibc-2.26/signal/../sysdeps/unix/sysv/linux/raise.c:48
abort at /build/glibc-itYbWN/glibc-2.26/stdlib/abort.c:90
__assert_fail_base at /build/glibc-itYbWN/glibc-2.26/assert/assert.c:92
__assert_fail at /build/glibc-itYbWN/glibc-2.26/assert/assert.c:101
jl_deserialize_typemap_entry at /home/keno/julia/src/dump.c:1886
jl_deserialize_value at /home/keno/julia/src/dump.c:2107
```

**Comment by Keno** [comment]
Note that this is actually while precompiling DataFrames which itself is de-serializing StatsBase.

**Comment by maleadt** [comment]
https://github.com/JuliaLang/julia/issues/26278#issuecomment-402733955 ?

**Comment by Keno** [comment]
Ah yes, looks like the same bug.

**Comment by Keno** [comment]
In any case:
```
(rr) p /x te->func.linfo->max_world
$2 = 0xffffffffffffffff
(rr) p /x te->func.linfo->min_world
$3 = 0x6bc5
(rr) p /x te->min_world
$4 = 0x1
(rr) p /x te->max_world
$5 = 0x0
```
so the typemap entry was deleted, but the corresponding linfo was not. cc @vtjnash who added this assertion.

**Comment by Keno** [comment]
Ok, I think the assertion was just wrong. Fixing that gets us to another assertion that @JeffBezanson says he has a fix to.

**Comment by JeffBezanson** [comment]
Fix for that is in #28216.
